helper/metricsutil/gauge_process_test.go

```diff
@@ -0,0 +1,490 @@
+package metricsutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/armon/go-metrics"
+	log "github.com/hashicorp/go-hclog"
+)
+
+// SimulatedTime maintains a virtual clock so the test isn't
+// dependent upon real time.
+// Unfortunately there is no way to run these tests in parallel
+// since they rely on the same global timeNow function.
+type SimulatedTime struct {
```

Comment:
Would it be worth adding SimulatedTime and clock/defaultClock to an sdk/helper package?